google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/util/AppEngineUtil.java

```diff
@@ -113,13 +124,50 @@ private AppEngineUtil() {
       }
     }
 
-    // I kind of expected the logic for adding the user specific deployment source to be in the
-    // artifact based deployment sources.
     moduleDeploymentSources.add(createUserSpecifiedPathDeploymentSource(project));
 
     return moduleDeploymentSources;
   }
 
+  /**
+   * Determines if a project is set up like an App Engine standard project but is configured
+   * in 'compatibility' mode. This indicates that the project runs in the flexible environment.
+   */
+  public static boolean isFlexCompat(@NotNull Project project, @NotNull DeploymentSource source) {
+    Artifact artifact;
+    if (source instanceof AppEngineArtifactDeploymentSource) {
+      artifact = ((AppEngineArtifactDeploymentSource) source).getArtifact();
+      if (artifact != null) {
+        return isFlexCompat(project, artifact);
+      }
+    }
+
+    return false;
+  }
+
+  private static boolean isFlexCompat(@NotNull Project project,
```

Comment:
I'd need to look into this. Currently in order to even create that kind of deployment source we need to first determine if the artifact is flex-compat. So doing that would require other refactoring.